fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowColumnsCommand.java

```diff
@@ -0,0 +1,256 @@
+    /**
+     * replaceColumnNameVisitor
+     * replace column name to real column name
+     */
+    private static class ReplaceColumnNameVisitor extends DefaultExpressionRewriter<Void> {
+        @Override
+        public Expression visitUnboundSlot(UnboundSlot slot, Void context) {
+            String name = slot.getName().toLowerCase(Locale.ROOT);
+            switch (name) {
+                case "field":
+                    return UnboundSlot.quoted("COLUMN_NAME");
+                case "type":
+                    return UnboundSlot.quoted("COLUMN_TYPE");
+                case "null":
+                    return UnboundSlot.quoted("IS_NULLABLE");
+                case "default":
+                    return UnboundSlot.quoted("COLUMN_DEFAULT");
+                case "comment":
+                    return UnboundSlot.quoted("COLUMN_COMMENT");
+                default:
+                    return slot;
+            }
+        }
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        if (whereClause != null) {
+            Expression rewritten = whereClause.accept(new ReplaceColumnNameVisitor(), null);
+            String whereCondition = " WHERE TABLE_NAME = '" + tableNameInfo.getTbl() + "' AND " + rewritten.toSql();
+            TableNameInfo info = new TableNameInfo(tableNameInfo.getCtl(), "information_schema", "columns");
+
+            List<AliasInfo> selectList = new ArrayList<>();
+            if (isFull) {
+                selectList.add(AliasInfo.of("COLUMN_NAME", "Field"));
+                selectList.add(AliasInfo.of("COLUMN_TYPE", "Type"));
+                selectList.add(AliasInfo.of("COLLATION_NAME", "Collation"));
+                selectList.add(AliasInfo.of("IS_NULLABLE", "Null"));
+                selectList.add(AliasInfo.of("COLUMN_KEY", "Key"));
+                selectList.add(AliasInfo.of("COLUMN_DEFAULT", "Default"));
+                selectList.add(AliasInfo.of("EXTRA", "Extra"));
+                selectList.add(AliasInfo.of("PRIVILEGES", "Privileges")); // optional, can be set to ''
+                selectList.add(AliasInfo.of("COLUMN_COMMENT", "Comment"));
+            } else {
+                selectList.add(AliasInfo.of("COLUMN_NAME", "Field"));
+                selectList.add(AliasInfo.of("COLUMN_TYPE", "Type"));
+                selectList.add(AliasInfo.of("IS_NULLABLE", "Null"));
+                selectList.add(AliasInfo.of("COLUMN_KEY", "Key"));
+                selectList.add(AliasInfo.of("COLUMN_DEFAULT", "Default"));
+                selectList.add(AliasInfo.of("EXTRA", "Extra"));
+            }
+
+            LogicalPlan plan = Utils.buildLogicalPlan(selectList, info, whereCondition);
+            List<List<String>> rows = Utils.executePlan(ctx, executor, plan);
+            for (List<String> row : rows) {
+                String rawType = row.get(1);
+                row.set(1, normalizeSqlColumnType(rawType));
```

Comment:
I am not sure why we need normalize column type, have you tried other data types like varchar, map, struct. etc...